lib/cisco_node_utils/router_ospf_area.rb

```diff
@@ -0,0 +1,285 @@
+
+    # CLI can be the following or none
+    # area 1.1.1.1 filter-list route-map bbb out
+    def filter_list_out
+      config_get('ospf_area', 'filter_list_out', @get_args)
+    end
+
+    def filter_list_out=(val)
+      return if filter_list_out == false && val == false
+      state = val ? '' : 'no'
+      rm = val ? val : filter_list_out
+      set_args_keys(state: state, route_map: rm)
+      config_set('ospf_area', 'filter_list_out', @set_args)
+    end
+
+    def default_filter_list_out
+      config_get_default('ospf_area', 'filter_list_out')
+    end
+
+    # range can take multiple values for the same vrf
+    # area 1.1.1.1 range 10.3.0.0/32
+    # area 1.1.1.1 range 10.3.0.1/32
+    # area 1.1.1.1 range 10.3.3.0/24
+    # area 1.1.1.1 range 10.3.0.0/16 not-advertise cost 23
+    # sometimes the not-advertise and cost are reversed in the
+    # show command for reasons unknown!
+    # area 1.1.1.1 range 10.3.0.0/16 cost 23 not-advertise
+    # use positional way of getting the values as it is
+    # simple and there are only two properties which are
+    # optional. ip is mandatory
+    # the return list is of the form [ip, not-advertise, cost]
+    # ex: [['10.3.0.0/16', true, '23'], ['10.3.0.0/32', true, false],
+    #      ['10.3.0.1/32', false, false], ['10.3.3.0/24', false, '450']]
+    def range
+      list = []
+      ranges = config_get('ospf_area', 'range', @get_args)
+      ranges.each do |line|
+        llist = []
+        params = line.split
+        llist[0] = params[0]
+        line.include?('not-advertise') ? llist[1] = true : llist[1] = false
+        if line.include?('cost')
+          llist[2] = params[params.index('cost') + 1]
+        else
+          llist[2] = false
+        end
+        list << llist
+      end
+      list
+    end
+
+    def range=(set_list)
+      # fail if set_list contains duplicate ip values
+      ip_list = []
+      set_list.each do |ip, _not_advertise, _cval|
+        ip_list << ip
+      end
+      fail ArgumentError, 'Duplicate ip values for range' unless
+        ip_list.size == ip_list.uniq.size
+      # reset the current ranges first due to bug CSCuz98937
+      cur_list = range
+      cur_list.each do |ip, _not_advertise, _cval|
+        set_args_keys(state: 'no', ip: ip, not_advertise: '',
+                      cost: '', value: '')
+        config_set('ospf_area', 'range', @set_args)
+      end
+      # now set the range from the set_list
+      set_list.each do |ip, not_advertise, cval|
+        na = not_advertise ? 'not-advertise' : ''
+        cost = cval ? 'cost' : ''
+        value = cval ? cval : ''
+        set_args_keys(state: '', ip: ip, not_advertise: na,
+                      cost: cost, value: value)
+        config_set('ospf_area', 'range', @set_args)
+      end
+    end
+
+    def default_range
+      config_get_default('ospf_area', 'range')
+    end
+
+    # CLI can be either of the following or none
+    # area 1.1.1.1 stub
+    # area 1.1.1.1 stub no-summary
```

Comment:
It's similar but it doesn't work for me in this case, since the default stub behavior is to summarize,  which will be confusing in a manifest. Let's reconsider this one.